services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/events/EventsCollectorImpl.java

```diff
@@ -91,11 +96,41 @@ private <E extends SynapseEvent> void log(List<E> events) {
 
 		recordsMap.forEach((streamName, records) -> {
 			if (!records.isEmpty()) {
+				logToJsonStream(streamName,records);
 				firehoseLogger.logBatch(streamName, records);
 			}
 		});
 	}
 
+	private void logToJsonStream(String streamName, List<AwsKinesisLogRecord> records) {

```

Comment:
This logic is tied to the file download/upload, it does not belong in this class.